src/main/java/hudson/plugins/audit_trail/AuditTrailPlugin.java

```diff
@@ -23,159 +23,146 @@
+import jenkins.model.GlobalConfiguration;
 import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 
+import javax.annotation.PostConstruct;
 import javax.servlet.ServletException;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
 
 /**
  * Keep audit trail of particular Jenkins operations, such as configuring jobs.
+ *
  * @author Alan Harder
+ * @author Pierre Beitz
  */
-public class AuditTrailPlugin extends Plugin {
+@Symbol("audit-trail")
+@Extension
+public class AuditTrailPlugin extends GlobalConfiguration {
+
+    private static final Logger LOGGER = Logger.getLogger(AuditTrailPlugin.class.getName());
     private String pattern = ".*/(?:configSubmit|doDelete|postBuildResult|enable|disable|"
-      + "cancelQueue|stop|toggleLogKeep|doWipeOutWorkspace|createItem|createView|toggleOffline|"
-      + "cancelQuietDown|quietDown|restart|exit|safeExit)";
+            + "cancelQueue|stop|toggleLogKeep|doWipeOutWorkspace|createItem|createView|toggleOffline|"
+            + "cancelQuietDown|quietDown|restart|exit|safeExit)";
     private boolean logBuildCause = true;
 
-    private List<AuditLogger> loggers = new ArrayList<AuditLogger>();
+    private List<AuditLogger> loggers = new ArrayList<>();
 
     private transient boolean started;
 
     private transient String log;
-    private transient int limit = 1, count = 1;
 
     public String getPattern() { return pattern; }
     public boolean getLogBuildCause() { return logBuildCause; }
     public List<AuditLogger> getLoggers() { return loggers; }
 
-    @Override public void start() throws Exception {
-        // Set a default value; will be overridden by load() once customized:
+    public AuditTrailPlugin() {
         load();
-        applySettings();
-
-        // Add Filter to watch all requests and log matching ones
-        PluginServletFilter.addFilter(new AuditTrailFilter(this));
     }
 
-    @Override public void configure(StaplerRequest req, JSONObject formData)
-            throws IOException, ServletException, FormException {
-        pattern = formData.optString("pattern");
-        logBuildCause = formData.optBoolean("logBuildCause", true);
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
         // readResolve makes sure loggers is initialized, so it should never be null.
+        // TODO this should probably be moved somewhere else
         loggers.forEach(AuditLogger::cleanUp);
-        loggers = Descriptor.newInstancesFromHeteroList(
-                req, formData, "loggers", getLoggerDescriptors());
-        save();
+        try (BulkChange bc = new BulkChange(this)) {
```

Comment:
what's the advantage of using bulkchange  here?
I would have  though just bindJSON would be enough?

Reply:
you're right there is no point in doing a bulk of...one change :)
I'll modify this.